tests/run-make/c-unwind-abi-catch-lib-panic/rmake.rs

```diff
@@ -0,0 +1,36 @@
+// Exercise unwinding a panic. This catches a panic across an FFI (foreign function interface)
+// boundary and downcasts it into an integer.
+// The Rust code that panics is in a separate crate.
+// See https://github.com/rust-lang/rust/commit/baf227ea0c1e07fc54395a51e4b3881d701180cb
+
+//@ ignore-cross-compile
+// Reason: the compiled binary is executed
+//@ needs-unwind
+// Reason: this test exercises unwinding a panic
+
+use run_make_support::{cc, is_msvc, llvm_ar, run, rustc};
+
+fn main() {
+    // Compile `add.c` into an object file.
+    if is_msvc() {
+        cc().arg("-c").out_exe("add").input("add.c").run();
+    } else {
+        cc().arg("-v").arg("-c").out_exe("add.o").input("add.c").run();
+    };
+
+    // Compile `panic.rs` into an object file.
+    // Note that we invoke `rustc` directly, so we may emit an object rather
+    // than an archive. We'll do that later.
+    rustc().emit("obj").input("panic.rs").run();
```

Comment:
Question: does this need to use `bare_rustc`?

Reply:
To my knowledge, no. `rustc` in `run_make_support` is a lot more "bare" than `tools.mk`'s `RUSTC`. In this test, they used `BARE_RUSTC` to be able to pass their own `out-dir`, but this is irrelevant in our rewrite since we did away with the `TMPDIR` spam in the tests themselves.